### PR TITLE
Add pybind11 demo wrapper for DateTime

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

File: python/lsst/daf/base/exampleLib.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from _exampleLib import DateTime
+
+def toPython(self, timescale=None):
+    """Convert a DateTime to Python's datetime
+
+    @param timescale  Timescale for resultant datetime
+    """
+    import datetime
+    nsecs = self.nsecs(timescale) if timescale is not None else self.nsecs()
+    return datetime.datetime.utcfromtimestamp(nsecs/10**9)
```
Comment (ktlim) [inline python/lsst/daf/base/exampleLib.py]: Isn't it dangerous to use utcfromtimestamp() on something that could be TAI?

File: python/lsst/daf/base/exampleLib.cc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    py::enum_<DateTime::DateSystem>(cls, "DateSystem")
+        .value("JD", DateTime::DateSystem::JD)
+        .value("MJD", DateTime::DateSystem::MJD)
+        .value("EPOCH", DateTime::DateSystem::EPOCH)
+        .export_values();
+
+    cls.def(py::init<long long, DateTime::Timescale>(),
+            py::arg("nsecs") = 0LL, py::arg("scale") = DateTime::TAI)
+        .def(py::init<long long, DateTime::Timescale>())
+        .def(py::init<double, DateTime::DateSystem, DateTime::Timescale>())
```
Comment (ktlim) [inline python/lsst/daf/base/exampleLib.cc]: Should we make it a standard practice to have Python-friendly docstrings here?

Comment (TallJimbo) [inline python/lsst/daf/base/exampleLib.cc]: IMO: eventually, yes, but @pschella probably isn't the right person to write most of these, and I don't think now is the time to do it.

File: python/lsst/daf/base/exampleLib.py (hunk 1 above)
Comment (timj) [inline python/lsst/daf/base/exampleLib.py]: When the time comes to do this properly we should just use `astropy.time` and https://github.com/lsst-dm/python-experiments/blob/master/lsstx/DateTime.py will be a useful resource.

**Pull request merged**